device/tt_cluster_descriptor.h

```diff
@@ -77,6 +77,7 @@ class tt_ClusterDescriptor {
   std::vector<std::tuple<ethernet_channel_t, ethernet_channel_t>> get_directly_connected_ethernet_channels_between_chips(const chip_id_t &first, const chip_id_t &second) const;
   
   bool is_chip_mmio_capable(const chip_id_t &chip_id) const;
+  bool is_chip_remote(const chip_id_t &chip_id) const;
```

Comment:
nit: consider passing by value - const reference can sometimes imply the object is large or expensive to copy.  (I do see that you're following the pattern of the method above...)